src/Library/Plugin/Page.php

```diff
@@ -0,0 +1,228 @@
+<?php
+/**
+ * BoldGrid Library Plugin Page.
+ *
+ * @package Boldgrid\Plugin
+ *
+ * @since SINCEVERSION
+ * @author BoldGrid <wpb@boldgrid.com>
+ */
+
+namespace Boldgrid\Library\Library\Plugin;
+
+/**
+ * Generic page class.
+ *
+ * This class represents a specific page used by the 
+ * Boldgrid\Library\Library\Plugin\Plugin class.
+ *
+ * @since SINCEVERSION
+ */
+
+class Page {
+
+    /**
+     * Plugin 
+     * 
+     * The Boldgrid\Library\Library\Plugin\Plugin Object
+     * that this page belongs to.
+     * 
+     * @var Plugin
+     * @access protected
+     */
+    protected $plugin;
+
+    /**
+     * Page Slug
+     * 
+     * @var string
+     * @access protected
+     * @since SINCEVERSION
+     */
+    protected $slug;
+    
+    /**
+     * Plugin Config
+     * 
+     * The Config array passed to this page
+     * 
+     * @var array
+     * @since SINCEVERSION
+     */
+    protected $pluginConfig;
```

Comment:
Could we skip having a `$pluginConfig` property and instead use the `$plugin` property like `$this->plugin->getConfigs()` ?